--- Python-2.7.2/Modules/_multiprocessing/multiprocessing.c.orig	2011-06-11 17:46:27.000000000 +0200
+++ Python-2.7.2/Modules/_multiprocessing/multiprocessing.c 		2012-11-01 01:34:34.172297156 +0100
@@ -122,7 +122,7 @@
     cmsg->cmsg_type = SCM_RIGHTS;
     cmsg->cmsg_len = CMSG_LEN(sizeof(int));
     msg.msg_controllen = cmsg->cmsg_len;
-    *CMSG_DATA(cmsg) = fd;
+    *(int*)CMSG_DATA(cmsg) = fd;
 
     Py_BEGIN_ALLOW_THREADS
     res = sendmsg(conn, &msg, 0);
@@ -165,7 +165,7 @@
     if (res < 0)
         return PyErr_SetFromErrno(PyExc_OSError);
 
-    fd = *CMSG_DATA(cmsg);
+    fd = *(int*)CMSG_DATA(cmsg);
     return Py_BuildValue("i", fd);
 }
 
